plugins/modules/cluster_info.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Ansible Cloud Team (@ansible-collections)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: cluster_info
+short_description: Gathers information about one or more clusters
+description:
+    - >-
+      Gathers information about one or more clusters.
+      You can search for clusters based on the cluster name, datacenter name, or a combination of the two.
+author:
+    - Ansible Cloud Team (@ansible-collections)
+
+options:
+    cluster:
+        description:
+            - The name of the cluster on which to gather info.
+            - At least one of O(datacenter) or O(cluster) is required.
+        type: str
+        required: false
+        aliases: [cluster_name, name]
+    datacenter:
+        description:
+            - The name of the datacenter.
+            - At least one of O(datacenter) or O(cluster) is required.
+        type: str
+        required: false
+        aliases: [datacenter_name]
+    gather_tags:
+        description:
+            - If true, gather any tags attached to the cluster(s)
+            - This has no affect if the O(schema) is set to V(vsphere). In that case, add 'tag' to O(properties) or leave O(properties) unset.
+        type: bool
+        default: false
+        required: false
+    schema:
+        description:
+            - Specify the output schema desired.
+            - The V(summary) output schema is the legacy output from the module.
+            - The V(vsphere) output schema is the vSphere API class definition.
+        choices: ['summary', 'vsphere']
+        default: 'summary'
+        type: str
+    properties:
+        description:
+            - If the schema is 'vsphere', gather these specific properties only
+        type: list
+        elements: str
+
+extends_documentation_fragment:
+    - vmware.vmware.vmware.vcenter_documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather Cluster Information
+  vmware.vmware.cluster_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    datacenter_name: datacenter
+    cluster_name: my_cluster
+  register: _out
+
+- name: Gather Information About All Clusters In a Datacenter
+  vmware.vmware.cluster_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    datacenter_name: datacenter
+  register: _out
+
+- name: Gather Specific Properties About a Cluster
```

Comment:
If there are multiple datacenters what will it return? the first one? please explain
I'm not sure why to use it without a DC name...